lgc/patch/PatchResourceCollect.cpp

```diff
@@ -1134,9 +1134,9 @@ void PatchResourceCollect::processShader() {
   }
 
   if (m_shaderStage == ShaderStageFragment) {
-    if (m_resUsage->builtInUsage.fs.fragCoord || m_resUsage->builtInUsage.fs.pointCoord ||
-        m_resUsage->builtInUsage.fs.sampleMaskIn) {
-      if (m_pipelineState->getRasterizerState().perSampleShading)
+    if (m_pipelineState->getRasterizerState().perSampleShading) {
+      if (m_resUsage->builtInUsage.fs.fragCoord || m_resUsage->builtInUsage.fs.pointCoord ||
+          m_resUsage->builtInUsage.fs.sampleMaskIn || m_resUsage->resourceWrite)
```

Comment:
I don't follow this hunk. Why is `m_resUsage->resourceWrite` added into the condition?

Reply:
In theory, as long as `perSampleShading` is true, we should enable sample shading. However, for the performance, we try not to enable it as much as possible. If `m_resUsage->resourceWrite` is true,  whether to enable sample shading has an impact on the output. Please see above shader. Thanks.